src/com/owncloud/android/utils/UploadUtils.java

```diff
@@ -0,0 +1,68 @@
+package com.owncloud.android.utils;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo.State;
+import android.os.BatteryManager;
+
+import com.owncloud.android.db.UploadDbObject;
+
+
+public class UploadUtils {
+
+    public static boolean isCharging(Context context) {
+        IntentFilter ifilter = new IntentFilter(Intent.ACTION_BATTERY_CHANGED);
+        Intent batteryStatus = context.registerReceiver(null, ifilter);
+
+        int status = batteryStatus.getIntExtra(BatteryManager.EXTRA_STATUS, -1);
+        return status == BatteryManager.BATTERY_STATUS_CHARGING || status == BatteryManager.BATTERY_STATUS_FULL;
+    }
+
+    public static boolean isOnline(Context context) {
+        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+        return cm.getActiveNetworkInfo() != null && cm.getActiveNetworkInfo().isConnected();
+    }
+
+    public static boolean isConnectedViaWiFi(Context context) {
+        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+        return cm != null && cm.getActiveNetworkInfo() != null
+                && cm.getActiveNetworkInfo().getType() == ConnectivityManager.TYPE_WIFI
+                && cm.getActiveNetworkInfo().getState() == State.CONNECTED;
+    }
+    
+    /**
+     * Returns true when user is able to cancel this upload. That is, when
+     * upload is currently in progress or scheduled for upload.
+     */
+    static public  boolean userCanCancelUpload(UploadDbObject uploadFile) {
```

Comment:
This method fits better in ```UploadDbObject```, and not as a ```static```. Its response is fully determined by the state of the argument ```uploadFile```, and it's public.